### ISSUE TITLE
Proper language settings/files

### ISSUE THREAD

**Issue by moonlightz** [eggheads/eggdrop#1220]
Hello... 
I am a Eggdrop user for at least 14 years and something that amused me is the fact that Eggdrop doesn't rely on xx_XX for a language setting, such as en_US, fr_FR, es_MX, etc...
I wonder if Eggdrop will use such kind of language definition in a near future and if the Eggdrop team will throw all strings to a translation service such as Crowdin or Transifex so us, regular users, can help translate and correct things... because, I being portuguese of Portugal, I see not only a full localized Eggdrop but I also see a mix of pt_PT and pt_BR strings in *.portuguese.* files. It is either duck or chicken.  One taste is different from the other.
I searched "translation" using the github box and found a thread from an user trying to fix transfers module language and we can see such language mixture.
Currently, I have 3 bots using English language and one bot having portuguese set as language. I tried myself correct strings but I give up because some are ok, some are not ok, and some are missing and the corrected parts vanishing with a bot upgrade.
The proper language scheme, like en_US, en_GB, de_DE, it_IT, ...., should have been used since long time ago, instead of english, french, italian,..... That's my opinion and an idea. I am not sure if someone asked anywhere about this.

**Comment by vanosg** [comment]
Hi, thanks for creating this issue! Yes, localization is an issue. We can certainly look at naming schemes, but I want to make sure you don't think that you can't contribute- if you have some changes you have made, we will GLADLY accept them right here via GitHub. You can submit a pull request with a new, localized language file, and we can add it to the eggdrop core repository. Having a Portugal- and Brazil-based language pack would be greatly desired! Please feel free to submit a PR here that has a localized language file and we'll get it incorporated. That is the first step; naming schema changes, if any, can come later.

Thanks for pointing out those localization websites, do you have experience with either of them? It would be interesting to see if that adds an incentive for people to add new languages. We haven't had a lot of interest from the userbase over the last few decades to add more; it is truly a tedious task. 

So, bottom line- if you are interested, please open a PR here and we'll figure out a way to get the Portugal dialect of Portuguese added.  Thanks!

**Comment by ZarTek-Creole** [comment]
Personally I use https://poedit.net/ for software translations in gettext. I think it would be nice to work on a proper language definition in eggdrop. It's work, it takes time but nothing very difficult I think?

**Comment by vanosg** [comment]
You're absolutely right, but time is the issue for the dev team. I think this issue would be a great starter issue for someone who wanted to contribute to Eggdrop is a very substantial way. If anyone is interested in undertaking an endeavor that could address this, please contact us on #eggdrop on Libera and we'd be happy to answer questions, as well as guide and mentor someone through the process. We look forward to working with someone on this!

**Comment by moonlightz** [comment]
Thanks for the positive reply. Yes, we all know it will not be easy.
I do use Crowdin and Transferix to translate other prople's projects to my language. Translation is not easy and we need to know what's really there and transmit to the end-user the proper meaning. 
I can try to PR and see what happens.

**Comment by vanosg** [comment]
Please do keep in touch with us in the channel; As this is a significant change, I'd hate for any work to be done that wouldn't be compatible with the direction Eggdrop is going. Thanks!
